src/encoded/static/components/cart/clear.js

```diff
@@ -89,67 +89,61 @@ CartClearModal.contextTypes = {
 /**
  * Display an actuator button to clear the current cart, and handle the resulting warning modal.
  */
-class CartClearButtonComponent extends React.Component {
-    constructor() {
-        super();
-        this.state = {
-            /** True if modal about clearing the cart is visible */
-            modalOpen: false,
-        };
-        this.handleClearCartClick = this.handleClearCartClick.bind(this);
-        this.handleCloseClick = this.handleCloseClick.bind(this);
-    }
+const CartClearButtonComponent = ({ elements, inProgress, isCartReadOnly }) => {
+    const [modalOpen, setModalOpen] = React.useState(false);
 
     /**
      * Handle a click in the Clear Cart button by showing the confirmation modal.
      */
-    handleClearCartClick() {
-        this.setState({ modalOpen: true });
-    }
+    const handleClearCartClick = () => {
```

Comment:
I keep changing how I name event handlers, but I think “ClearCart” helps here instead of just “Clear” which might still leave people wondering “Clear what? The cart? A field?”

Reply:
Got point.